### PR TITLE
Remove persistance and fix Codacy issues

### PR DESCRIPTION
**Pull request by Awarua-**
Completes #111

### CODE REVIEW

File: core/core_integrations/slack.js [hunk 1]
```diff
@@ -605,7 +585,7 @@ exports.stop = function() {
 			while (numSocketsToShutDown > 0 && !clearedMessageQueue && !clearedTimeouts) {
 				sleep.usleep(500000);
 			}
-			callback(null, "finished");
+			callback(null, 'finished');
 		}
 	], function(err, results) {
```
Comment (mrkno) [inline core/core_integrations/slack.js]: Codacy Issue found: ['results' is defined but never used.](https://www.codacy.com/app/theprojectshop/Kassy/file/2549394133/issues/source?bid=3282987&fileBranchId=3288593#l590)

File: core/core_integrations/slack.js [hunk 2]
```diff
@@ -128,31 +128,30 @@ renameChannel = function(title, thread) {
...
+			type:'typing',
+			channel: teamInfo.channel_id
 		};
 
 	body.id = getNextMessageId(teamId);
 	body = JSON.stringify(body);
 
-	console.debug("slack-> send typing indication");
+	console.debug('slack-> send typing indication');
 	sendSocketMessage(teamId, body);
```
Comment (mrkno) [inline core/core_integrations/slack.js]: Codacy Issue found: ["sendSocketMessage" was used before it was defined](https://www.codacy.com/app/theprojectshop/Kassy/file/2549394133/issues/source?bid=3282987&fileBranchId=3288593#l147)

File: core/core_integrations/slack.js [hunk 3]
```diff
@@ -17,77 +17,77 @@ inTransaction = {},
...
 	sendMessageWebAPI(teamId);
 },
 
-sendMessageWebAPI = function (teamId) {
-	// Initialise for team
-	if (inTransaction[teamId] === null || inTransaction[teamId] === undefined) {
-		inTransaction[teamId] = false;
-    }
+sendMessage = function(message, thread) {
+	var teamInfo = getChannelIdAndTeamId(thread);
```
Comment (mrkno) [inline core/core_integrations/slack.js]: Codacy Issue found: ["getChannelIdAndTeamId" was used before it was defined](https://www.codacy.com/app/theprojectshop/Kassy/file/2549394133/issues/source?bid=3282987&fileBranchId=3288593#l65)

File: core/core_integrations/slack.js (hunk 1 above)
Comment (mrkno) [inline core/core_integrations/slack.js]: Codacy Issue found: ["results" is defined but never used](https://www.codacy.com/app/theprojectshop/Kassy/file/2549394133/issues/source?bid=3282987&fileBranchId=3288593#l590)

File: core/core_integrations/slack.js [hunk 4]
```diff
@@ -17,77 +17,77 @@ inTransaction = {},
...
-tempTeamData = {},
+teamData = {},
 
-sendMessage = function(message, thread) {
-	var teamInfo = getChannelIdAndTeamId(thread);
+sendMessageWebAPI = function (teamId) {
+	// Initialise for team
+	if (inTransaction[teamId] === null) {
+		inTransaction[teamId] = false;
+    }
```
Comment (mrkno) [inline core/core_integrations/slack.js]: weird indentation

File: core/core_integrations/slack.js [hunk 5]
```diff
@@ -217,33 +234,55 @@ init = function(token, callback) {
...
+	team.pongTimeout = setTimeout(function() {
+		if (team.pong_id === pongId && !shuttingDown) {
+			console.debug('slack-> terminating connection as socket failed to respond to ping request.');
+			sockets[teamId].close();
+		}
+	}, defaultPingResponseTimeout);
+},
+
 startPingPongTimer = function(teamId) {
 	if (!Date.now) {
```
Comment (mrkno) [inline core/core_integrations/slack.js]: Wait... why are we messing with built in JS prototypes?

File: core/core_integrations/slack.js [hunk 6]
```diff
@@ -605,8 +584,9 @@ exports.stop = function() {
 			while (numSocketsToShutDown > 0 && !clearedMessageQueue && !clearedTimeouts) {
 				sleep.usleep(500000);
```
Comment (mrkno) [inline core/core_integrations/slack.js]: Given this is slack related, do you mind fixing the `npm install sleep` problems on Windows while we are here.

File: core/core_integrations/grunt.js [hunk 7]
```diff
@@ -100,31 +105,38 @@ exports.start = function(cb) {
...
+        socket.on('close', function() {
+            console.debug('Client disconnected');
+        });
     });
 
-    socket.on('error', function(error) {
+    wss.on('error', function(error) {
         console.error(error);
-    });
+    })
```
Comment (mrkno) [inline core/core_integrations/grunt.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/theprojectshop/Kassy/file/2571707052/issues/source?bid=3282987&fileBranchId=3288593#l134)

Comment (mrkno) [inline core/core_integrations/grunt.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/theprojectshop/Kassy/file/2571707052/issues/source?bid=3282987&fileBranchId=3288593#l134)

File: core/core_integrations/slack.js [hunk 8]
```diff
@@ -17,596 +15,553 @@ inTransaction = {},
...
-		});
-	}
-	else {
-		console.debug("slack-> Whoops looks like you don't have the slack config set up correctly");
-	}
+    var slackTokens = exports.config.slack_tokens;
+
+    shuttingDown = false;
+    eventReceivedCallback = callback;
+    numSocketsToShutDown = 0;
```
Comment (mrkno) [inline core/core_integrations/slack.js]: Codacy Issue found: [Variable 'numSocketsToShutDown' implicitly declared.](https://www.codacy.com/app/theprojectshop/Kassy/file/2571707054/issues/source?bid=3282987&fileBranchId=3288593#l533)

File: core/core_integrations/slack.js [hunk 9]
```diff
@@ -17,596 +15,553 @@ inTransaction = {},
...
-			teamRename(event, teamId);
-			break;
-		case 'pong':
-			pong(event, teamId);
-			break;
-		default:
-			console.debug("slack-> Message of type " + event.type + " not supported");
-			break;
-	}
+findSlackBot = function(body, slackTeam) {
```
Comment (mrkno) [inline core/core_integrations/slack.js]: Codacy Issue found: ['findSlackBot' is defined but never used.](https://www.codacy.com/app/theprojectshop/Kassy/file/2571707054/issues/source?bid=3282987&fileBranchId=3288593#l302)

File: core/core_integrations/slack.js [hunk 10]
```diff
@@ -17,596 +15,553 @@ inTransaction = {},
...
+    if (socket !== null) {
+        switch(type) {
+            case 'messsage':
+                socket.send(body);
+                break;
+            case 'ping':
+                socket.ping();
+                break;
+            case 'pong':
+                socket.pong();
```
Comment (mrkno) [inline core/core_integrations/slack.js]: Codacy Issue found: [Expected indentation of 12 space characters but found 16.](https://www.codacy.com/app/theprojectshop/Kassy/file/2571707054/issues/source?bid=3282987&fileBranchId=3288593#l144)

File: core/core_integrations/slack.js [hunk 11]
```diff
@@ -1,12 +1,10 @@
 var request = require.safe('request'),
 shim = require.once('../shim.js'),
 WebSocket = require.safe('ws'),
-async = require.safe('async'),
-sleep = require.safe('sleep'),
+deasync = require.safe('deasync'),
 platform = null,
 sockets = [],
```
Comment (mrkno) [inline core/core_integrations/slack.js]: Codacy Issue found: [Expected indentation of 4 space characters but found 0.](https://www.codacy.com/app/theprojectshop/Kassy/file/2571707054/issues/source?bid=3282987&fileBranchId=3288593#l6)

File: core/core_integrations/slack.js [hunk 12]
```diff
@@ -17,596 +15,553 @@ inTransaction = {},
...
-
-			tempTeamData[teamId] = {};
-
-			// find slackbot
-			findSlackBot(body, team);
-			callback({"team_id": teamId, "url": body.url});
-		}
-	});
-},
+generateChannelsMap = function(data) {
```
Comment (mrkno) [inline core/core_integrations/slack.js]: Codacy Issue found: [Expected indentation of 4 space characters but found 0.](https://www.codacy.com/app/theprojectshop/Kassy/file/2571707054/issues/source?bid=3282987&fileBranchId=3288593#l177)

File: core/core_integrations/slack.js [hunk 13]
```diff
@@ -17,596 +15,553 @@ inTransaction = {},
...
+    var thread_components = thread.split('\0', 2),
+    channelId = thread_components[0],
+    teamId = thread_components[1],
+    token = teamData[teamId].token;
+
+    return { channel_id: channelId, team_id: teamId, token: token };
+},
 
-	return { "channel_id": channelId, "team_id": teamId, "token": token };
+sendMessage = function(message, thread) {
```
Comment (mrkno) [inline core/core_integrations/slack.js]: Codacy Issue found: [Expected indentation of 4 space characters but found 0.](https://www.codacy.com/app/theprojectshop/Kassy/file/2571707054/issues/source?bid=3282987&fileBranchId=3288593#l71)

File: core/core_integrations/slack.js [hunk 14]
```diff
@@ -17,596 +15,553 @@ inTransaction = {},
...
-
-	body.id = getNextMessageId(teamId);
-	body = JSON.stringify(body);
-
-	console.debug("slack-> send typing indication");
-	sendSocketMessage(teamId, body);
+sendSocketMessage = function(type, teamId, body) {
+    var socket = sockets[teamId];
+    if (socket !== null) {
+        switch(type) {
```
Comment (mrkno) [inline core/core_integrations/slack.js]: Codacy Issue found: [Keyword "switch" must be followed by whitespace.](https://www.codacy.com/app/theprojectshop/Kassy/file/2571707054/issues/source?bid=3282987&fileBranchId=3288593#l136)

File: core/core_integrations/slack.js [hunk 15]
```diff
@@ -17,596 +15,553 @@ inTransaction = {},
...
+            team.token = token;
+            team.event_id = 1;
+            team.lastMessageSinceConnection = false;
+
+            // Generate user map
+            team.users = generateUserMap(body);
+            team.allChannels = generateChannelsMap(body);
+
+            // find slackbot
+            findSlackBot(body, team);
```
Comment (mrkno) [inline core/core_integrations/slack.js]: Codacy Issue found: ["findSlackBot" was used before it was defined](https://www.codacy.com/app/theprojectshop/Kassy/file/2571707054/issues/source?bid=3282987&fileBranchId=3288593#l250)

File: .travis.yml [hunk 16]
```diff
@@ -1,4 +1,7 @@
 language: node_js
+os:
+    - linux
+    - osx
```
Comment (mrkno) [inline .travis.yml]: no windows test? Also these changes are unrelated to this branch (leave them cause they are small though)

Comment (Awarua-) [inline .travis.yml]: Windows not supported by Travis CI, yeah I know should be separate branches

**Comment by mrkno** [comment]
Approved
